src/screens/ProfileScreen/index.js

```diff
@@ -91,29 +97,45 @@ const Header = (props) => {
     setTabIndexToSigned,
     setTabIndexToAnonymous
   } = props;
-
+  const navigator = useNavigation();
+  const [feedsContext, dispatch] = React.useContext(Context).feeds;
+
+  const {feeds, timer, viewPostTimeIndex} = feedsContext;
+  // eslint-disable-next-line no-underscore-dangle
+  const handleOnAddPostButtonClicked = () => {
+    const currentTime = new Date().getTime();
+    const id = feeds && feeds[viewPostTimeIndex]?.id;
+    if (id) viewTimePost(id, currentTime - timer.getTime(), SOURCE_FEED_TAB);
+    navigator.navigate(NavigationConstants.CREATE_POST_SCREEN);
+    setTimer(new Date(), dispatch);
+  };
   return (
     <View
       onLayout={(event) => {
         const headerHeightLayout = event.nativeEvent.layout.height;
         headerHeightRef.current = headerHeightLayout;
       }}>
       <View style={styles.content}>
-        <View style={{flexDirection: 'row', alignContent: 'center', alignItems: 'center'}}>
+        <View style={{flexDirection: 'row'}}>
           <ProfilePicture
             onImageContainerClick={changeImage}
             profilePicPath={dataMain.profile_pic_path}
             karmaScore={dataMain.karma_score}
             withKarma={true}
+            size={!dataMain.is_karma_unlocked ? 48 : undefined}
+            width={!dataMain.is_karma_unlocked ? 4 : undefined}
           />
           <View
             style={{
               flexDirection: 'column',
               paddingHorizontal: 14,
-              paddingVertical: 5,
               justifyContent: 'center'
             }}>
-            <KarmaScore score={Math.floor(dataMain.karma_score)} />
+            {dataMain.is_karma_unlocked ? (
+              <KarmaScore score={Math.floor(dataMain.karma_score)} />
+            ) : (
+              <KarmaLock onPressCreatePost={handleOnAddPostButtonClicked} />
+            )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The profile picture rendering logic has been updated. Ensure that this change is thoroughly tested across different screen sizes and resolutions to ensure the UI layout remains consistent.
```diff
+ <View
+   onLayout={(event) => {
+     const headerHeightLayout = event.nativeEvent.layout.height;
+     headerHeightRef.current = headerHeightLayout;
+   }}>
+   <View style={styles.content}>
+    <View style={{flexDirection: 'row'}}>
+       <ProfilePicture
+         onImageContainerClick={changeImage}
+         profilePicPath={dataMain.profile_pic_path}
+         karmaScore={dataMain.karma_score}
+         withKarma={true}
+        size={!dataMain.is_karma_unlocked ? 48 : undefined}
+        width={!dataMain.is_karma_unlocked ? 4 : undefined}
+       />
+       <View
+         style={{
+           flexDirection: 'column',
+           paddingHorizontal: 14,
+           justifyContent: 'center'
+         }}>
+        {dataMain.is_karma_unlocked ? (
+          <KarmaScore score={Math.floor(dataMain.karma_score)} />
+        ) : (
+          <KarmaLock onPressCreatePost={handleOnAddPostButtonClicked} />
+        )}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->